internal/pkg/pkg.go

```diff
@@ -0,0 +1,53 @@
+// Package pkg defines methods and heuristics for sanitizing package output.
+package pkg
+
+import (
+	"github.com/replit/upm/internal/api"
+)
+
+func makeLoweredHM(b api.LanguageBackend, names []string) map[api.PkgName]struct{} {
+	// Build a hashset. struct{}{} purportedly is of size 0, so this is as good as we get
+	set := make(map[api.PkgName]struct{})
+	for _, pkg := range names {
+		normal := b.NormalizePackageName(api.PkgName(pkg))
+		set[normal] = struct{}{}
+	}
+	return set
+}
+
+func SortPackages(b api.LanguageBackend, query string, ignoredPackages []string, packages []api.PkgInfo) []api.PkgInfo {
+
+	ignoredPackageSet := makeLoweredHM(b, ignoredPackages)
```

Comment:
I don't think this initialization loop is necessary. You just want a map of bool, I think.